composition-js/src/composeDirectiveManager.ts

```diff
@@ -143,6 +143,7 @@ export class ComposeDirectiveManager {
           this.pushHint(new CompositionHint(
             HINTS.DIRECTIVE_COMPOSITION_INFO,
             `Non-composed core feature "${coreIdentity}" has major version mismatch across subgraphs`,
+            coreIdentity,
```

Comment:
It looks like our testing story is a bit inconsistent, which is not your fault.

i.e. if I change this to, say, `undefined`, tests still pass. But if I remove the `this.pushHint(...)` there is a test that fails (which doesn't use the `toRaiseHint` matcher.

I'd like it if we addressed these inconsistencies so there's coverage for all of your additions. Let me know if you'd like me to help with that, since I think that's an "us" problem.

Reply:
Note: this is the only one I checked, so for now I'm not checking the others or leaving comments on them.

Reply:
Pretty sure if we get the testing story consistent it'll reveal the directive name errors that I pointed out above (or reveal that we're just lacking coverage on those particular hints altogether).